kratos/sources/model_part.cpp

```diff
@@ -855,6 +855,291 @@ void ModelPart::RemoveElementsFromAllLevels(Flags identifier_flag)
+{
+
+    KRATOS_TRY
+    if (IsSubModelPart())
+    {
+        MasterSlaveConstraintType::Pointer p_new_constraint = mpParentModelPart->CreateNewMasterSlaveConstraint(ConstraintName, Id, rMasterNode,
+                                                                                                                    rMasterVariable,
+                                                                                                                    rSlaveNode,
+                                                                                                                    rSlaveVariable,
+                                                                                                                    Weight,
+                                                                                                                    Constant);
+        this->AddMasterSlaveConstraint(p_new_constraint);
+        return p_new_constraint;
+    }
+
+    auto existing_element_iterator = this->mMasterSlaveConstraints.find(Id);
+    if(existing_element_iterator != this->MasterSlaveConstraintsEnd() )
+        KRATOS_ERROR << "trying to construct an master-slave constraint with ID " << Id << " however a constraint with the same Id already exists";
+
+
+    //create the new element
+    MasterSlaveConstraintType const& r_clone_constraint = KratosComponents<MasterSlaveConstraintType>::Get(ConstraintName);
+    MasterSlaveConstraintType::Pointer p_new_constraint = r_clone_constraint.Create(Id, rMasterNode,
+                                                                                        rMasterVariable,
+                                                                                        rSlaveNode,
+                                                                                        rSlaveVariable,
+                                                                                        Weight,
+                                                                                        Constant);
+
+    //add the new element
+    this->AddMasterSlaveConstraint(p_new_constraint);
+
+    return p_new_constraint;
+    KRATOS_CATCH("")
+
+}
+
+
+/** Remove the master-slave constraint with given Id from mesh with ThisIndex in this modelpart and all its subs.
+*/
+void ModelPart::RemoveMasterSlaveConstraint(ModelPart::IndexType MasterSlaveConstraintId)
+{
+    mMasterSlaveConstraints.erase(MasterSlaveConstraintId);
+
+    for (SubModelPartIterator i_sub_model_part = SubModelPartsBegin(); i_sub_model_part != SubModelPartsEnd(); i_sub_model_part++)
+        i_sub_model_part->RemoveMasterSlaveConstraint(MasterSlaveConstraintId);
+}
+
+/** Remove given master-slave constraint from mesh with ThisIndex in this modelpart and all its subs.
+*/
+void ModelPart::RemoveMasterSlaveConstraint(ModelPart::MasterSlaveConstraintType& ThisMasterSlaveConstraint)
+{
+    mMasterSlaveConstraints.erase(ThisMasterSlaveConstraint.Id());
```

Comment:
You can avoid code redundancy by doing `RemoveMasterSlaveConstraint(ThisMasterSlaveConstraint.Id());`